Switch to Token from Token Protocol

AfreenLikeCaffeine opened the pull request:
In this PR:

- Removed TokenProtocol and switched to using Token type for generating graph client
- Consolidated test files and greatly reduced tests.
- Lots of code removed for simplication
- Update dependencies and versions

Commit:
feat!: rearchitect graph package to use Token type instead of TokenProtocol

BREAKING CHANGES:
- get_graph_client() now accepts Token (str | StringLike | TokenFactory | None) instead of Callable[[], TokenProtocol]
- Removed TokenProtocol and protocols.py entirely
- Added dependency on microsoft-teams-common package for unified Token type
- Simplified authentication error handling - no custom validation
- Removed token caching in DirectTokenCredential

Migration Guide:
Before: get_graph_client(lambda: MyTokenClass('token'))
After:  get_graph_client('token') or get_graph_client(lambda: 'token')

Features:
- Support for string tokens, callables, async callables, and None
- Unified Token type across Teams AI SDK packages
- Proper async/sync context handling with ThreadPoolExecutor
- Comprehensive test coverage (26 tests)
- Updated documentation and examples

Files changed:
- graph.py: Updated signature to accept Token instead of Callable[[], TokenProtocol]
- auth_provider.py: Complete rewrite using resolve_token() utility
- protocols.py: DELETED - removed TokenProtocol entirely
- pyproject.toml: Added microsoft-teams-common dependency
- README.md: Updated with new usage patterns and examples
- tests/: Comprehensive test updates for Token approach

Commit:
refactor: remove transitive dependencies from graph package

- Removed azure-identity (transitive via msgraph-sdk)
- Removed cryptography (transitive via azure-identity)
- Removed PyJWT[crypto] (not used - Teams API handles token validation)

Only keeping directly used dependencies:
- azure-core (for TokenCredential interface)
- msgraph-sdk (core Graph client)
- microsoft-teams-common (Token type and resolve_token)

All 26 tests still pass, functionality unchanged.

Commit:
feat: add version constraints to graph package dependencies

- azure-core>=1.31.0 (minimum required by msgraph-sdk's azure-identity dependency)
- msgraph-sdk>=1.30.0,<2.0.0 (allow patch/minor updates, prevent breaking major version changes)
- microsoft-teams-common>=1.0.0 (ensure stable API compatibility)

Version constraints provide:
- Minimum compatibility requirements
- Protection against breaking changes in major versions
- Clear dependency expectations for package consumers

Commit:
Minor string fix

Commit:
Consolidate duplicate tests and clean up dependencies

- Merged test_graph_token.py into test_graph.py to eliminate ~80% test duplication
- Added connection_name parameter testing to consolidated tests
- Removed unused dotenv dependency from tests/graph/pyproject.toml
- All 17 tests pass with comprehensive coverage maintained

Commit:
Improve code organization and simplify async usage

- Move concurrent.futures import to top level in auth_provider.py
- Convert get_graph_client from async to sync function (no async operations needed)
- Update all tests to remove unnecessary @pytest.mark.asyncio and await calls
- Update documentation examples to reflect synchronous API
- Update application code to use simplified non-async get_graph_client
- All 17 tests pass with improved performance and cleaner API

Commit:
Improve token resolution timeout management

- Add _TOKEN_RESOLUTION_TIMEOUT class constant (5 seconds)
- Replace hardcoded 30-second timeout with documented constant
- Reduce timeout to more reasonable 5 seconds for token operations
- Improve code maintainability with clear timeout documentation
- All tests pass with improved timeout handling

Commit:
Refactor: Rename DirectTokenCredential to AuthProvider and improve JWT handling

- Rename DirectTokenCredential class to AuthProvider for clearer naming
- Extract JWT expiration time when available instead of using default
- Add PyJWT dependency for proper JWT token parsing
- Implement configurable timeout constant (5 seconds)
- Consolidate duplicate test files and update test class names
- Simplify graph.py from async to sync function
- Clean up dependencies and organize imports
- All 19 tests passing with improved functionality

Addresses code review feedback for better naming conventions and
JWT expiration handling.

Commit:
Minor comment fix

heyitsaamir reviewed (approved):
Minor comments, but lgtm!

Commit:
Merge branch 'main' into afreenlikecaffeine/token

Commit:
Optimize dependencies and simplify graph client usage

- Update PyJWT dependency to match api package: pyjwt[crypto]>=2.10.0
- Fix microsoft-teams-common version to match actual package: 0.0.1-alpha.1
- Simplify graph test app to use ctx.user_token instead of manual API calls
- Remove unused GetUserTokenParams import from test app
- Maintain all functionality while reducing code complexity

Addresses code review feedback for cleaner token handling.

Commit:
Remove unused connection_name parameter from AuthProvider and get_graph_client

- Remove connection_name parameter from AuthProvider.__init__()
- Remove connection_name parameter from get_graph_client() function
- Simplify API interface to focus on core token functionality
- Update all tests to use simplified API
- Update README.md with cleaner usage examples
- Update test app to use simplified get_graph_client() call
- Remove unused Optional import from auth_provider.py

All 19 tests passing with simplified, cleaner API.

Commit:
Fix type system inconsistency: make token parameter optional with proper None handling

- Update get_graph_client signature to Optional[Token] = None
- Add immediate None validation with clear error message
- Remove timeout parameter from AuthProvider as requested in code review
- Update tests to validate new None handling behavior
- Addresses feedback from liyydu about Token type allowing None but runtime rejecting it

All 19 tests passing ✅

lilyydu reviewed (approved):
LGTM! 🥳

Commit:
Organize imports: move all imports to top level

- Move time and jwt imports from inline to top level
- Remove redundant inline imports already available at top level
- Follow PEP 8 import organization conventions
- All 19 tests still passing ✅

Improves code readability and follows Python best practices.

Commit:
Add logging for token expiration fallback scenarios

- Add debug logging when JWT token is missing 'exp' claim
- Add debug logging when token is not a valid JWT
- Both scenarios now log when falling back to 1-hour default expiration
- Helps with debugging token authentication issues
- Update .gitignore to exclude .copilot-instructions.md

All 19 tests still passing ✅
